chef/cookbooks/ntp/recipes/default.rb

```diff
@@ -29,6 +29,10 @@
 if node["roles"].include?("ntp-server")
   ntp_servers += node[:ntp][:external_servers]
   is_server = true
+  listen_network_addresses = (node[:ntp][:listen_networks] || []).map do |network|
+    Barclamp::Inventory.get_network_by_type(node, network).address
```

Comment:
You're assuming that the node has an address allocate for that network, but you're not ensuring this in `apply_role_pre_chef_call`.

Reply:
Do you think `apply_role_pre_chef_call` method should be doing the allocation or do you think it should be raising an error if an address isn't already allocated? This method is kind of mysterious to me.